enzy_htp/_interface/amber_interface.py

```diff
@@ -2834,6 +2834,79 @@ def remove_traj_solvent(
         contents = "\n".join(contents)
         self.run_cpptraj(contents)
 
+    def get_average_structure(
+            self,
+            traj_path: str,
+            prmtop_path: str,
+            auto_image: bool = False,
+        ) -> Structure:
+        """Calculate the average structure of a trajectory file.
+        
+        Args:
+            traj_path (str): The path to the .nc/.mdcrd file.
+            prmtop_path (str): The path to the prmtop file.
+            auto_image (bool, optional): Indicate if to move the structures from all frames together (to the box center).
+
+        Returns:
+            average_structure (Structure): A structure that represents the average coordinates of the entire trajectory.   
+        """
+        avg_structure_filename = os.path.join(eh_config.system.SCRATCH_DIR, "temp_traj_average.pdb")
+
+        contents: List[str] = [
+            f"parm {prmtop_path}",
+            f"trajin {traj_path}",
+        ]
+        if auto_image:
+            contents += [
+                "autoimage"
+            ]
+        contents += [
+            f"average {avg_structure_filename} pdb",
+            "quit",
+        ]
+        contents = "\n".join(contents)
+        self.run_cpptraj(contents)
+        average_structure = PDBParser().get_structure(avg_structure_filename)
+        
+        fs.safe_rm(avg_structure_filename)
+        return average_structure
+
+    def get_rmsd(
+            self,
+            traj_path: str,
+            prmtop_path: str,
+            mask_pattern: str = str(),
+        ) -> Structure:
+        """Calculate the RMSD value of a trajectory file.
+        
+        Args:
+            traj_path (str): The path to the .nc/.mdcrd file.
+            prmtop_path (str): The path to the prmtop file.
+            mask_pattern (str, optional): mask_pattern (str): A pymol-formatted selection string which defines the region for calculating RMSD value."""
```

Comment:
This wont work. Amber does not take the "pymol-formatted selection string". Also this API should not take the pattern as the input, but instead a StruSeletion() object.

Reply:
Resolved.